### DISCUSSION TITLE
How to get "instance type" in lua ?

### DISCUSSION THREAD

**Discussion by FabQi** [RomanKubiak/ctrlr#618] [Q&A]
Hello 

How can I get the panel "instance type" in lua ?
(I mean something like 'standalone', 'restricted', etc..)

Regards 
Fabrice

**Comment by damiensellier** [comment]
Can you please expand on why you need it? I don't think it's possible since the instance type property is not stored at the panel level but added to the prefs on export of the binaries.

**Reply by FabQi** [reply]
Hello

I'm trying to "protect" the instance. (if instanceType == "restricted" then.....)
I' don't know how, checking for (good) ideas...

Regards
Fabrice

**Reply by damiensellier** [reply]
What do you mean about protect? code sign? encrypt? unlock with serial codes for commercial registration? disable editing? set the exported bundle rights to read only?

To understand the restricted mode you have to understand at which point it's declared as such.
When you export an instance, the application is cloned to the destination location with the panel file included and the related external  user resources (images fonts etc) but during the copy some properties are changed to re identify the software as a new, different software from the panel name, manufacturer...
When you run this exported instance, it will automatically load the associated panel.

When you export as restricted there's a secret property that is writen to declare it as restricted.  When the exported instance is opened  with its included panel, the "Restricted" flag prevents the app to show the panel editor pane, some of the "unrestricted" functions accessible in the menu and more. Only showing the bare minimum for users only. It's ideal for commercial products for exemple or "read-only" distributable.

The architecture is pretty simple, and from the get go, the panel developer  will not export their final products as both restricted and "unrestricted" and won't need to use statements to differentiate if the app is running in restricted mode or not. You know it at the early stage of the production.
If you want to remove features still available even in restricted mode, just tell me and I'll add a parameter to enable/disable this feature in restrected mode from the core.
Je parle français si tu préfères donc hésites pas. Dominique @dobo365 aussi.
Damien
